.github/actions/generate-specs/split-tests.js

```diff
@@ -0,0 +1,96 @@
+class DeviceInfo {
+  constructor(device, osVersion) {
+    this.device = device;
+    this.osVersion = osVersion;
+  }
+}
+
+class SpecGroup {
+  constructor(runId, specs, deviceInfo) {
+    this.runId = runId;
+    this.specs = specs;
+    this.device = deviceInfo.device;
+    this.osVersion = deviceInfo.osVersion;
+  }
+}
+
+class Specs {
+  constructor(directory, searchPath, parallelism, deviceInfo) {
+    this.searchPath = searchPath;
+    this.directory = directory;
+    this.parallelism = parallelism;
+    this.rawFiles = [];
+    this.groupedFiles = [];
+    this.deviceInfo = deviceInfo;
+  }
+
+  findFiles() {
+    const fs = require('fs');
+    const path = require('path');
+    const dirPath = path.join(this.directory, this.searchPath);
```

Comment:
Since this is being joined and I don't see them used separately, instead of having two separate env's (`DIRECTORY` and `SEARCH_PATH`), can we use one env only? Maybe `TEST_DIRECTORY` with default value of `detox/e2e/test`?